### PR TITLE
Create cron task to calculate filedir size and generate a report

### PR DESCRIPTION
**Pull request by gleimermora**

### CODE REVIEW

File: classes/local/store/object_file_system.php [hunk 1]
```diff
@@ -806,4 +806,33 @@ public function add_file_from_string($content) {
 
         return $result;
     }
+
+    /**
+     * Returns the total size of the filedir directory in bytes.
+     * @return float|int
+     */
+    public function get_filedir_size() {
+        $output = $this->exec_command('/usr/bin/du -sk ' . escapeshellarg($this->filedir) . ' | cut -f1');
```
Comment (brendanheywood) [inline classes/local/store/object_file_system.php]: can you please use the $CFG->pathtodu cfg variable for this

File: db/tasks.php [hunk 2]
```diff
@@ -89,5 +89,14 @@
         'dayofweek' => '*',
         'month'     => '*'
     ),
+    [
+        'classname' => 'tool_objectfs\task\check_filedir_size',
+        'blocking'  => 0,
+        'minute'    => '*',
+        'hour'      => '2',
```
Comment (brendanheywood) [inline db/tasks.php]: This doesnt seem right, this is every minute from 2am to 3am? Did you intend just once at 2am each night?

File: classes/local/store/object_file_system.php (hunk 1 above)
Comment (gleimermora) [reply classes/local/store/object_file_system.php]: updated

File: db/tasks.php (hunk 2 above)
Comment (gleimermora) [reply db/tasks.php]: updated

File: classes/local/report/location_report_builder.php [hunk 3]
```diff
@@ -69,8 +72,20 @@ public function build_report() {
...
+            $row = end($filedirrows);
+            $deltacount = $row->objectcount - $totalcount;
+            $deltasize = $row->objectsum - $totalsum;
+            $key = 'deltaa';
+            if ($totalsum > $row->objectsum) {
+                $deltacount = $totalcount - $row->objectcount;
+                $deltasize = $totalsum - $row->objectsum;
+                $key = 'deltab';
+            }
+            $report->add_row($key, $deltacount, $deltasize);
```
Comment (brendanheywood) [inline classes/local/report/location_report_builder.php]: can you add this row regardless please?

bonus points, add a link here to scheduled task gui to update the stats to the latest

Comment (gleimermora) [reply classes/local/report/location_report_builder.php]: updated

File: renderer.php [hunk 4]
```diff
@@ -66,22 +69,50 @@ private function render_location_report($report) {
...
+        $runnabletasks = tool_task\run_from_cli::is_runnable();
+        $task = \core\task\manager::get_scheduled_task($classname);
+        if (!$task->get_disabled() && get_config('tool_task', 'enablerunnow') && $runnabletasks) {
+            $link = html_writer::link(
+                new moodle_url(
+                    '/admin/tool/task/schedule_task.php',
+                    ['task' => 'tool_objectfs\task\check_filedir_size']
+                ),
+                get_string('object_status:filedir:update', 'tool_objectfs'),
+                ['style' => 'font-size:0.75rem;']
```
Comment (brendanheywood) [inline renderer.php]: a simple trick I use it just using a <small>blah</small> for stuff like this

**Pull request merged**